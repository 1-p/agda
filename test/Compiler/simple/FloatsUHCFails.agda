-- ASR (2016-09-16). These tests are disabled with the UHC
-- backend. After fixing a test to merge it into the `Floats` tests.

module _ where

open import Common.Prelude

print : Float → IO Unit
print x = putStrLn (primShowFloat x)

pi : Float
pi = 3.141592653589793

main : IO Unit
main =
  -- See issues #1856 and #1857.
  putStr "√2 = " ,, print (primFloatSqrt 2.0) ,,
  putStr "√2 = " ,, print (primFloatTimes 2.0 (primSin (primFloatDiv pi 4.0))) ,,
  return unit
