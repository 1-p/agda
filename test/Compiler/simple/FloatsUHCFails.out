EXECUTED_PROGRAM

ret > ExitSuccess
out > √2 = 1.4142135623730951
out > √2 = 1.414213562373095
out > Inf == Inf  = true
out > NaN == NaN  = true
out > 0.0 == -0.0 = false
out > isZero  0.0 = pos
out > isZero -0.0 = neg
out > isZero  1.0 = nonzero
out > e = 2.718281828459045
out >
